cmd/osv-scanner/main.go

```diff
@@ -35,211 +34,149 @@ func run(args []string, stdout, stderr io.Writer) int {
-					return fmt.Errorf("--experimental-licenses requires at least one value")
-				}
-				if unrecognized := spdx.Unrecognized(allowlist); len(unrecognized) > 0 {
-					return fmt.Errorf("--experimental-licenses requires comma-separated spdx licenses. The following license(s) are not recognized as spdx: %s", strings.Join(unrecognized, ","))
-				}
-			}
-
-			verbosityLevel, err := reporter.ParseVerbosityLevel(context.String("verbosity"))
-			if err != nil {
-				return err
-			}
-
-			if r, err = reporter.New(format, stdout, stderr, verbosityLevel, termWidth); err != nil {
-				return err
-			}
-
-			var callAnalysisStates map[string]bool
-			if context.IsSet("experimental-call-analysis") {
-				callAnalysisStates = createCallAnalysisStates([]string{"all"}, context.StringSlice("no-call-analysis"))
-				r.Infof("Warning: the experimental-call-analysis flag has been replaced. Please use the call-analysis and no-call-analysis flags instead.\n")
-			} else {
-				callAnalysisStates = createCallAnalysisStates(context.StringSlice("call-analysis"), context.StringSlice("no-call-analysis"))
-			}
-
-			vulnResult, err := osvscanner.DoScan(osvscanner.ScannerActions{
-				LockfilePaths:        context.StringSlice("lockfile"),
-				SBOMPaths:            context.StringSlice("sbom"),
-				DockerContainerNames: context.StringSlice("docker"),
-				Recursive:            context.Bool("recursive"),
-				SkipGit:              context.Bool("skip-git"),
-				NoIgnore:             context.Bool("no-ignore"),
-				ConfigOverridePath:   context.String("config"),
-				DirectoryPaths:       context.Args().Slice(),
-				CallAnalysisStates:   callAnalysisStates,
-				ExperimentalScannerActions: osvscanner.ExperimentalScannerActions{
-					LocalDBPath:    context.String("experimental-local-db-path"),
-					CompareLocally: context.Bool("experimental-local-db"),
-					CompareOffline: context.Bool("experimental-offline"),
-					// License summary mode causes all
-					// packages to appear in the json as
-					// every package has a license - even
-					// if it's just the UNKNOWN license.
-					ShowAllPackages: context.Bool("experimental-all-packages") ||
-						context.Bool("experimental-licenses-summary"),
-					ScanLicensesSummary:   context.Bool("experimental-licenses-summary"),
-					ScanLicensesAllowlist: context.StringSlice("experimental-licenses"),
-				},
-			}, r)
-
-			if err != nil && !errors.Is(err, osvscanner.VulnerabilitiesFoundErr) {
-				return err
-			}
-
-			if errPrint := r.PrintResult(&vulnResult); errPrint != nil {
-				return fmt.Errorf("failed to write output: %w", errPrint)
+	}
+	allCommands := getAllCommands(app.Commands)
+	if len(args) >= 2 {
+		// Insert the default command to args if no command is specified.
+		if !slices.Contains(allCommands, args[1]) {
+			// Avoid in-place change to args as it is not a pointer.
+			argsTmp := make([]string, len(args)+1)
+			copy(argsTmp[2:], args[1:])
+			argsTmp[1] = app.DefaultCommand
+			args = argsTmp
```

Comment:
I think this can be
```go
args = slices.Insert(args, 1, app.DefaultCommand)
```

But I have a question: shouldn't setting `app.DefaultCommand` handle this? or does it just not work as we want it to?

Reply:
`app.DefaultCommand` only works when we don't pass any other args in. Once we pass other args, like flags or filename, it will treat them as a command. 
Some of our UTs are using the same args to run multiple times and expect the same results, so I didn't do in-place change here.